pymeasure/instruments/hp/hp4145x.py

```diff
@@ -0,0 +1,1252 @@
+        Set the instrument to automatically calibrate internally every 15minutes.
+        """,
+        values=[True, False],
+        validator=strict_discrete_set,
+        check_set_errors=True
+    )
+
+    def self_test(self):
+        """
+        Execute the self test and return pass/fail.
+        """
+        self.write("SF")
+        status = self.status
+        return (not (status & Status.SELF_TEST_FAIL)) and (status & Status.END_STATUS)
+
+    def stop(self):
+        """Stop the ongoing measurement.
+
+        .. code-block:: python
+
+            instr.stop()
+        """
+        self.write("MD ME4")
+
+    def measure(self, continuous=False):
+        """
+        Perform a measurement.
+
+        .. code-block::python
+
+            instr.measure() # for normale measurement (single)
+            instr.measure(continuous=True) # for repeated infinite measurement
+        """
+        if continuous:
+            self.write("MD ME2")
+        else:
+            self.write("MD ME1")
+
+    def disable_all(self):
+        """ Disable all channels in the instrument.
+
+        .. code-block:: python
+
+            instr.disable_all()
+        """
+        self.smu1.reset_settings()
+        self.smu2.reset_settings()
+        self.smu3.reset_settings()
+        self.smu4.reset_settings()
+        self.vmu1.reset_settings()
+        self.vmu2.reset_settings()
+
+    def configure(self, config_file):
+        """ Configure the channel setup and sweep using a JSON configuration file.
+
+        (JSON is the `JavaScript Object Notation`_)
+
+        .. _`JavaScript Object Notation`: https://www.json.org/
+
+        :param config_file: JSON file to configure instrument channels.
+
+        .. code-block:: python
+
+            instr.configure('config.json')
+        """
+        self.disable_all()
+        obj_dict = {'SMU1': self.smu1,
+                    'SMU2': self.smu2,
+                    'SMU3': self.smu3,
+                    'SMU4': self.smu4,
+                    'VMU1': self.vmu1,
+                    'VMU2': self.vmu2,
+                    'VSU1': self.vsu1,
+                    'VSU2': self.vsu2,
+                    'VAR1': self.var1,
+                    'VAR2': self.var2,
+                    'VARD': self.vard
+                    }
+        with open(config_file) as stream:
+            try:
+                instr_settings = json.load(stream)
+            except json.JSONDecodeError as e:
+                print(e)
+
+        # replace dict keys with Instrument objects
+        new_settings_dict = {}
+        for key, value in instr_settings.items():
+            new_settings_dict[obj_dict[key]] = value
+
+        for obj, setup in new_settings_dict.items():
+            for setting, value in setup.items():
+                setattr(obj, setting, value)
+                time.sleep(0.1)
+
+    def select_graphics_mode(self, mode, **kwargs):
+        """
+        Selects and configures the respective graphics mode.
```

Comment:
Docstrings in imperative mode:
```suggestion
        Select and configure the respective graphics mode.
```